python/pyarrow/parquet.py

```diff
@@ -1740,7 +1740,7 @@ def write_to_dataset(table, root_path, partition_cols=None,
     Parameters
     ----------
     table : pyarrow.Table
-    root_path : str,
+    root_path : str, pathlib.Path
```

Comment:
Ok, but we should probably disallow passing a `Path` if the filesystem is not a local filesystem.

Reply:
OK, and what about objects with `__fspath__` (the protocol doesn't seem to be explicitly only about local paths, but in practice might also be?)

Reply:
The same.